### ISSUE TITLE
Increased RateLimit (RequestLimitExceeded) running kops update at DescribeLaunchTemplates step

### ISSUE THREAD

**Issue by phspagiari** [kubernetes/kops#9806]
**1. What `kops` version are you running? The command `kops version`, will display
 this information.**
`Version 1.17.1 (git-4dfd4779fe)`
**2. What Kubernetes version are you running? `kubectl version` will print the
 version if a cluster is running or provide the Kubernetes version specified as
 a `kops` flag.**
`Server Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.11", GitCommit:"ea5f00d93211b7c80247bf607cfa422ad6fb5347", GitTreeState:"clean", BuildDate:"2020-08-13T15:11:47Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}`

**3. What cloud provider are you using?**
`AWS`

**4. What commands did you run?  What is the simplest way to reproduce this issue?**
`kops -v10 update cluster $NAME 2>&1 | tee -a log`

**5. What happened after the commands executed?**
We see kops Executing Tasks getting LaunchTemplates and returing the results but in the sequence we can see a lot of requests without response and a lot of `RequestLimitExceeded` from `ec2/DescribeLaunchTemplateVersions`. We are getting RateLimit for around 10 minutes to run a simple update.

**6. What did you expect to happen?**
Run kops update without `RateLimit` / `RequestLimitExceeded` from AWS.

**7. Please provide your cluster manifest. Execute
  `kops get --name my.example.com -o yaml` to display your cluster manifest.
  You may want to remove your cluster name and other sensitive information.**

```yaml
apiVersion: kops.k8s.io/v1alpha2
kind: Cluster
metadata:
  creationTimestamp: null
  generation: 6
  name: CLUSTER-NAME
spec:
  additionalPolicies:
    node: |
      [
        {
          "Effect": "Allow",
          "Action": ["sts:AssumeRole"],
          "Resource": ["*"]
        },
        {
          "Effect": "Allow",
          "Action": ["autoscaling:DescribeAutoScalingGroups","autoscaling:DescribeAutoScalingInstances","autoscaling:DescribeTags","autoscaling:SetDesiredCapacity","autoscaling:TerminateInstanceInAutoScalingGroup"],
          "Resource": ["*"]
        }
      ]
  api:
    loadBalancer:
      type: Internal
      useForInternalApi: true
  authorization:
    rbac: {}
  channel: stable
  cloudLabels:
    App: k8s
    Env: prod
    KubernetesVersion: 1.17.11
    Provider: kops
  cloudProvider: aws
  configBase: s3://XXXXXX
  dnsZone: XXXXXXXX
  docker:
    insecureRegistry: http://registry-mirror.XXXXX
    mtu: 8912
    registryMirrors:
    - http://registry-mirror.XXXXXXXXX
  etcdClusters:
  - etcdMembers:
    - instanceGroup: master1
      name: 1
      volumeSize: 150
    - instanceGroup: master2
      name: 2
      volumeSize: 150
    - instanceGroup: master3
      name: 3
      volumeSize: 150
    name: main
    version: 3.2.24
  - etcdMembers:
    - instanceGroup: master1
      name: 1
    - instanceGroup: master2
      name: 2
    - instanceGroup: master3
      name: 3
    name: events
    version: 3.2.24
  fileAssets:
  - content: |
      { "clusters": [ ] }
    name: webhook.json
    path: /srv/kubernetes/webhook.json
    roles:
    - Master
  hooks:
  - disabled: true
    name: update-engine.service
  - disabled: true
    name: locksmithd.service
  - before:
    - update-engine.service
    manifest: |
      Type=oneshot
      ExecStart=/usr/bin/systemctl stop update-engine.service
    name: disable-update-engine.service
  - before:
    - locksmithd.service
    manifest: |
      Type=oneshot
      ExecStart=/usr/bin/systemctl stop locksmithd.service
    name: disable-locksmithd.service
  - manifest: |
      Type=oneshot
      ExecStart=/usr/sbin/modprobe br_netfilter
    name: fix-dns.service
  - manifest: |
      Type=oneshot
      ExecStart=/sbin/sysctl -w  net.netfilter.nf_conntrack_tcp_be_liberal=1
    name: fix-nf-conntrack-tcp-be-liberal
  iam:
    legacy: true
  kubeAPIServer:
    auditLogMaxAge: 10
    auditLogMaxBackups: 1
    auditLogMaxSize: 100
    auditLogPath: /var/log/kube-apiserver-audit.log
    auditPolicyFile: /srv/kubernetes/audit.yaml
    authenticationTokenWebhookConfigFile: /srv/kubernetes/webhook.json
    authorizationMode: RBAC
    featureGates:
      EndpointSlice: "true"
      ServiceTopology: "true"
    maxMutatingRequestsInflight: 400
    maxRequestsInflight: 800
    runtimeConfig:
      batch/v2alpha1: "true"
  kubeControllerManager:
    featureGates:
      EndpointSlice: "true"
      ServiceTopology: "true"
  kubeDNS:
    provider: None
  kubeProxy:
    featureGates:
      EndpointSlice: "true"
      ServiceTopology: "true"
      SupportIPVSProxyMode: "true"
    proxyMode: ipvs
  kubeScheduler:
    featureGates:
      EndpointSlice: "true"
      ServiceTopology: "true"
  kubelet:
    clusterDNS: 169.254.20.10
    evictionHard: memory.available<500Mi,nodefs.available<15%,nodefs.inodesFree<5%,imagefs.available<25%,imagefs.inodesFree<5%
    featureGates:
      EndpointSlice: "true"
      ServiceTopology: "true"
    kubeReserved:
      cpu: 250m
      ephemeral-storage: 1Gi
      memory: 500Mi
    kubeReservedCgroup: /kube-reserved
    systemReserved:
      cpu: 200m
      ephemeral-storage: 1Gi
      memory: 500Mi
    systemReservedCgroup: /system-reserved
  kubernetesApiAccess:
  - XXXXXXXXXXX
  kubernetesVersion: v1.17.11
  masterPublicName: XXXXXXXXXXX
  networkCIDR: XXXXXXX
  networkID: vpc-XXXXXXX
  networking:
    cni: {}
  nonMasqueradeCIDR: XXXXXXXXX/16
  sshAccess:
  - XXXXXXXX
  subnets:
  XXXXXXX

```

**8. Please run the commands with most verbose logging by adding the `-v 10` flag.
  Paste the logs into this report, or in a gist and provide the gist link here.**
```console
[...]
I0824 17:25:47.324417   54033 executor.go:176] Executing task "LaunchTemplate/nodes-data-spot-e-green.CLUSTERNAME": *awstasks.LaunchTemplate {"Name":"nodes-data-spot-e-green.CLUSTERNAME"," {...}}
I0824 17:25:47.324559   54033 executor.go:176] Executing task "LaunchTemplate/nodes-gpu-blue.CLUSTERNAME": *awstasks.LaunchTemplate {"Name":"nodes-gpu-blue.CLUSTERNAME", {...}}
I0824 17:25:47.327444   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplates
I0824 17:25:47.327543   54033 request_logger.go:45] AWS request: ec2/DescribeSecurityGroups
I0824 17:25:47.332690   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplates
I0824 17:25:47.327583   54033 request_logger.go:45] AWS request: ec2/DescribeSecurityGroups
I0824 17:25:47.331273   54033 request_logger.go:45] AWS request: ec2/DescribeSecurityGroups
I0824 17:25:47.332758   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplates
I0824 17:25:47.331290   54033 request_logger.go:45] AWS request: autoscaling/DescribeLaunchConfigurations
I0824 17:25:47.331344   54033 request_logger.go:45] AWS request: ec2/DescribeSecurityGroups
I0824 17:25:47.331351   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplates
I0824 17:25:47.331596   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplates
I0824 17:25:47.331665   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplates
I0824 17:25:47.331977   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplates
I0824 17:25:47.331994   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplates
I0824 17:25:47.332388   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplates
I0824 17:25:47.332404   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplates
I0824 17:25:47.332416   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplates
I0824 17:25:48.030318   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.049972   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.055977   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.060722   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.062979   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.065519   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.065661   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.067549   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.070519   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.073392   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.074077   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.075844   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.087951   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.091813   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.095181   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.095251   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.095843   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.098524   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.098582   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.101096   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.105423   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.109109   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.110630   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.110776   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.258997   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.260523   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.269770   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.270438   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.272859   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.273485   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.277380   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.280208   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.282665   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.289422   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.291683   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.292378   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.298983   54033 load_balancer.go:268] Querying ELB tags for [ ... ]
I0824 17:25:48.299117   54033 request_logger.go:45] AWS request: elasticloadbalancing/DescribeTags
I0824 17:25:48.305775   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.306621   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.307538   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.310232   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.324779   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.325401   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.325894   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.331260   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.337420   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.337770   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.339969   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.459479   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.475442   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.487740   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.488001   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.489062   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.491451   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.502954   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.502957   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.507767   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.513203   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.520203   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.520728   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.524003   54033 request_logger.go:45] AWS request: elasticloadbalancing/DescribeLoadBalancers
I0824 17:25:48.526273   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.530071   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.535324   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.544896   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.553568   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.553568   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.571939   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.572005   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.572656   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.572668   54033 request_logger.go:45] AWS request: autoscaling/DescribeLaunchConfigurations
I0824 17:25:48.572840   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.587636   54033 request_logger.go:45] AWS request: autoscaling/DescribeLaunchConfigurations
I0824 17:25:48.592226   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.599918   54033 request_logger.go:45] AWS request: autoscaling/DescribeLaunchConfigurations
I0824 17:25:48.650596   54033 request_logger.go:45] AWS request: autoscaling/DescribeLaunchConfigurations
I0824 17:25:48.656582   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.661315   54033 request_logger.go:45] AWS request: autoscaling/DescribeLaunchConfigurations
I0824 17:25:48.661315   54033 request_logger.go:45] AWS request: autoscaling/DescribeLaunchConfigurations
I0824 17:25:48.685308   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.699543   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.703823   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.721885   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.723368   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.723882   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.725546   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.732477   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.738251   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.739818   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.742015   54033 load_balancer.go:268] Querying ELB tags for [ ... ]
I0824 17:25:48.742153   54033 request_logger.go:45] AWS request: elasticloadbalancing/DescribeTags
I0824 17:25:48.746909   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.748910   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.750105   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.760748   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.761652   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.765512   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.773528   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.780013   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.797101   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.803956   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.805063   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.810118   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.824705   54033 request_logger.go:45] AWS request: autoscaling/DescribeLaunchConfigurations
I0824 17:25:48.831259   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.865545   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.903116   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.906708   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.911281   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.928062   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.928771   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.929614   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.930763   54033 load_balancer.go:268] Querying ELB tags for [ ... ]
I0824 17:25:48.930829   54033 request_logger.go:45] AWS request: elasticloadbalancing/DescribeTags
I0824 17:25:48.932157   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.943725   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.947506   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.963163   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.967937   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.968259   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.970461   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.973637   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.981811   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.984529   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:48.989375   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:49.015802   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:49.016299   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:49.018727   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:49.034310   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:49.043441   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:49.061700   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:49.104920   54033 request_logger.go:45] AWS request: elasticloadbalancing/DescribeLoadBalancerAttributes
I0824 17:25:49.108104   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:49.118348   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:49.119628   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:49.124222   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:49.140500   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:49.147837   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:49.148776   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:49.153147   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:49.154963   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:49.159792   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:49.187130   54033 request_logger.go:45] AWS request: autoscaling/DescribeLaunchConfigurations
I0824 17:25:49.191129   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:49.194381   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:49.199105   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:49.203204   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:49.204914   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:49.225939   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:49.226583   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:49.226764   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:49.235876   54033 request_logger.go:45] AWS request: autoscaling/DescribeLaunchConfigurations
I0824 17:25:49.241600   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:49.250779   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:49.251880   54033 request_logger.go:45] AWS request: ec2/DescribeLaunchTemplateVersions
I0824 17:25:49.302507   54033 request_logger.go:45] AWS request: autoscaling/DescribeLaunchConfigurations
W0824 17:25:49.307995   54033 retry_handler.go:99] Got RequestLimitExceeded error on AWS request (ec2::DescribeLaunchTemplateVersions)
W0824 17:25:49.307995   54033 retry_handler.go:99] Got RequestLimitExceeded error on AWS request (ec2::DescribeLaunchTemplateVersions)
I0824 17:25:49.308062   54033 logging_retryer.go:60] Retryable error (RequestLimitExceeded: Request limit exceeded.
	status code: 503, request id: d91bfda9-8d09-48b4-8340-2894fcaecfc2) from ec2/DescribeLaunchTemplateVersions - will retry after delay of 627.275414ms
I0824 17:25:49.308099   54033 logging_retryer.go:60] Retryable error (RequestLimitExceeded: Request limit exceeded.
	status code: 503, request id: c007b5dc-eebc-4ea8-b3ed-48cc85bbc461) from ec2/DescribeLaunchTemplateVersions - will retry after delay of 791.223227ms
W0824 17:25:49.308787   54033 retry_handler.go:99] Got RequestLimitExceeded error on AWS request (ec2::DescribeLaunchTemplateVersions)
W0824 17:25:49.308823   54033 retry_handler.go:99] Got RequestLimitExceeded error on AWS request (ec2::DescribeLaunchTemplateVersions)
I0824 17:25:49.308821   54033 logging_retryer.go:60] Retryable error (RequestLimitExceeded: Request limit exceeded.
	status code: 503, request id: 3ddde110-18b2-455d-95a2-a278c2c903fd) from ec2/DescribeLaunchTemplateVersions - will retry after delay of 810.901988ms
I0824 17:25:49.308868   54033 logging_retryer.go:60] Retryable error (RequestLimitExceeded: Request limit exceeded.
	status code: 503, request id: 31d4aa17-5d10-4b07-b36b-9e707833fecd) from ec2/DescribeLaunchTemplateVersions - will retry after delay of 686.247317ms
W0824 17:25:49.308841   54033 retry_handler.go:99] Got RequestLimitExceeded error on AWS request (ec2::DescribeLaunchTemplateVersions)
I0824 17:25:49.308897   54033 logging_retryer.go:60] Retryable error (RequestLimitExceeded: Request limit exceeded.
	status code: 503, request id: 6fdbf26f-f470-42a9-9205-bff8fd6f15f8) from ec2/DescribeLaunchTemplateVersions - will retry after delay of 726.996772ms
[...]
```

**9. Anything else do we need to know?**
I've figured out that we had a lot of unused LaunchTemplates at our AWS account... I don't know why kops isn't cleaning up, anyway, I did a manual cleanup but made no effect.

Another thing is: This is happening since `v1.17.0-beta1`.

**Comment by johngmyers** [comment]
When finding deletions, determining the latest version of a launchtemplate, etc. kops will enumerate all versions of all lauchtemplates, then filter that enumerated list for the ones that have the name that it is interested in.

**Comment by phspagiari** [comment]
Sorry @johngmyers I didn't get. Our cluster isn't big at all and is the same size since kops v1.14, we didn't changed the quantity of launch templates / launch configurations (old version) so why we started facing this? I also pointed to a lot of LaunchTemplate left behind that kops didn't deleted. 

I opened a case with AWS just to double check that this limit is a "Hard" limit for all AWS accounts so isn't a AWS problem. Another thing is: We only use LaunchTemplates for kops, we don't have any other tool / application using the Limit of this API, just kops... I'm really stucked.

**Comment by phspagiari** [comment]
#### More Info:
 - Our cluster are splitted in 30 instancegroups
 - We monitored in Cloud Trail 3568 Calls for `DescribeLaunchTemplateVersions` while running a single `kops update cluster` and we have only `42` Launch Templates.

**Comment by phspagiari** [comment]
Hey @johngmyers! I just tested your PR #9909 with our development cluster, and the RateLimit completely vanished from the logs. See a example:


### Building v1.17.2 + cherry-pick of your code
```
I0914 12:02:14.772105   34458 executor.go:103] Tasks: 114 done / 134 total; 17 can run
I0914 12:02:15.623769   34458 dnsname.go:111] AliasTarget for "XXXX." is "XXX."
I0914 12:02:15.951229   34458 dnsname.go:111] AliasTarget for "XXX" is "XXX."
I0914 12:02:16.460047   34458 executor.go:103] Tasks: 131 done / 134 total; 3 can run
I0914 12:02:16.887731   34458 executor.go:103] Tasks: 134 done / 134 total; 0 can run
Will modify resources:
[...]
Must specify --yes to apply changes
```

### V1.17.2 only

```
I0911 14:49:20.166213   26223 executor.go:103] Tasks: 0 done / 134 total; 49 can run
I0911 14:49:21.257352   26223 executor.go:103] Tasks: 49 done / 134 total; 34 can run
I0911 14:49:22.169088   26223 executor.go:103] Tasks: 83 done / 134 total; 31 can run
W0911 14:49:26.246884   26223 retry_handler.go:99] Got RequestLimitExceeded error on AWS request (ec2::DescribeLaunchTemplateVersions)
[...] (a lot of RequestLimitExceeded here)
W0911 14:51:42.239553   26223 retry_handler.go:99] Got RequestLimitExceeded error on AWS request (ec2::DescribeLaunchTemplateVersions)
I0911 14:51:49.125070   26223 executor.go:103] Tasks: 114 done / 134 total; 17 can run
```

**Comment by johngmyers** [comment]
Appears to have been sufficiently addressed by #9909 
/close
